### ISSUE TITLE
Can civetweb load cgi_interpreter from ENV

### ISSUE THREAD

**Issue by quantuumsnot** [civetweb/civetweb#854]
Yo, testing latest 1.12 and found that it cannot load "php-cgi.exe" as ENV. PHP is maintained by package manager for Windows (scoop in this case) and is runnable from everywhere. Should I setup the full path of the php-cgi.exe each time when PHP dir is changed ?

**Comment by bel2125** [comment]
I don't know "scoop".

To get PHP running, I downloaded PHP for Windows from https://windows.php.net/download#php-7.4
Then, I set `cgi_interpreter C:\Full\Path\To\php-7.4.4-Win32-vc15-x64\php-cgi.exe`
... this worked for me.

You want to set this `C:\Full\Path\To\php-7.4.4-Win32-vc15-x64\php-cgi.exe` from an environment variable?
You could start CivetWeb from the command line:
```
SET PHP="C:\Full\Path\To\php-7.4.4-Win32-vc15-x64\php-cgi.exe"
civetweb_lua.exe -cgi_interpreter %PHP%
```

**Comment by bel2125** [comment]
Did this work for you?
Is there anything else we can do here, or can we close this issue?

**Comment by quantuumsnot** [comment]
Nope, the idea was to start php-cgi.exe with custom ini added as parameter to 'cgi_interpreter box' in civetweb settings. For example `D:\scoop\apps\php\current\php-cgi.exe -c "D:\phpini\7.2.ini"`

**Comment by bel2125** [comment]
Would it work if you use an intermediat CMD file?

The content would be the PHP executable, the options `-c D:\phpine\72.ini` followed by `%1`.
Since I don't have a 72.ini at hand, I tested without:
```
type C:\path\to\php.cmd
C:\Full\Path\To\php-7.4.4-Win32-vc15-x64\php-cgi.exe %1
```

You can start it as before with the cmd file as interpreter:
```
SET PHPC="C:\path\to\php.cmd"
civetweb_lua.exe -cgi_interpreter %PHPC%
```

Please check if this works for you!
 
We can still think about adding some `cgi_interpreter_args` configuration value - this is fairly easy to implement, just up to now nobody asked for it. Currently there is no additional configuration value.

**Comment by quantuumsnot** [comment]
> We can still think about **adding some `cgi_interpreter_args` configuration value** - this is fairly easy to implement, just up to now nobody asked for it. Currently there is no additional configuration value.

Yes, that was my idea because calling external batch file (already doing it) is somewhat a lazy fix :)

**Comment by bel2125** [comment]
The additional option "cgi_interpreter_args" has been implemented.
However, I cannot test it, since I do not have an 72.INI file and a PHP script that would create a different result depending on 72.INI being loaded or not.

**Comment by quantuumsnot** [comment]
`7.2.ini` was an dummy nickname, used just for the example. Usually I use different config `ini` files for each PHP version

**Comment by bel2125** [comment]
In a test with an empty x.ini file, the result is, the x.php page is identical with/without x.ini argument.
Not a very meaningful test ... but I don't have a better test at hand - at least not for PHP.

**Comment by quantuumsnot** [comment]
[php.ini-production.ini-production.txt](https://github.com/civetweb/civetweb/files/4458698/php.ini-production.ini-production.txt)
[php.ini-production.ini-development.txt](https://github.com/civetweb/civetweb/files/4458699/php.ini-production.ini-development.txt)

For testing sse these two files from standard PHP 7.4.4 installation - they are supplied by PHP dev team

**Comment by bel2125** [comment]
With cgi_interpreter_args "C:\Source\github\civetweb\test\phptest\php.ini-production.ini-development" opening http://localhost:8080/phptest/phpbenchlab.php returns

Fatal error: Uncaught Error: Call to undefined function mb_internal_encoding() in C:\Source\github\civetweb\test\phptest\phpbenchlab.php:6 Stack trace: #0 {main} thrown in C:\Source\github\civetweb\test\phptest\phpbenchlab.php on line 6

Without: exactly the same

Line 6 is `mb_internal_encoding('UTF-8');`

If I comment this line ...


Fatal error: Uncaught Error: Call to undefined function mb_http_output() in C:\Source\github\civetweb\test\phptest\phpbenchlab.php:7 Stack trace: #0 {main} thrown in C:\Source\github\civetweb\test\phptest\phpbenchlab.php on line 7

Line 7 is `mb_http_output('UTF-8');`

If I comment this line too ...

could not find drivercould not find drivercould not find drivercould not find drivercould not find drivercould not find driver

+ some table with test cases

3 of them passed, 3 of them failed

Independent if cgi_interpreter_arg is set to a valid script, set to "bumlux" (some garbage) or not set at all.

I am not a PHP expert - I can run a test with an existing PHP file, I can read an understand simple existing PHP files - but I cannot create a test for an issue like this, or debug any problem in PHP.

**Comment by quantuumsnot** [comment]
Seems it's working. Where to download the exe?

**Comment by bel2125** [comment]
I usually only built an exe when a version is released.
I can provide a beta version for you to test
[civetweb64_085e10c309fe63b8e3f61efd3d58e7cc9e8f3051.zip](https://github.com/civetweb/civetweb/files/4463025/civetweb64_085e10c309fe63b8e3f61efd3d58e7cc9e8f3051.zip)

**Comment by quantuumsnot** [comment]
Ok, remove it - it works
